### ISSUE TITLE
cptbox multiprocess support

### ISSUE THREAD

**Issue by quantum5** [DMOJ/judge#203]
`cptbox`, after #188, should logically implement multiprocess debugging support. This may one day permit the implementation of a `BASH` language, if we manage to limit what it can `exec`.

**Comment by quantum5** [comment]
Now, everywhere we use `kill`, i.e. shocker, protection faults, we have to kill by `pgid`, which for `cptbox`-based process trees is `-pid` of the first child.

**Comment by quantum5** [comment]
Debian/kFreeBSD `jessie` does not have `PT_FOLLOW_FORK` defined. However, passing `23` works.

**Comment by Riolku** [comment]
Isn't this complete?

**Comment by Xyene** [comment]
No, this is not implemented in a safe manner yet, or we'd have shell executors on DMOJ :)

**Comment by Riolku** [comment]
Will landlock allow this to be cleaned up?

**Comment by Xyene** [comment]
Yes, that's one of the biggest missing pieces.

**Comment by Riolku** [comment]
Would we need to only allow bash when paired with landlock? I guess so... is the other option to stop all processes when a file access happens?

**Comment by Xyene** [comment]
Yes, and ensure that they actually all get stopped, which sounds painful. We'd probably want to have some non-bash-builtins too, but this is super low-priority so hasn't been given much thought.
